modules/ldap-syncing-rfc2307.adoc

```diff
@@ -0,0 +1,307 @@
+
+* How the group and users are added to the LDAP server.
+* What the resulting group record in {product-title} will be after synchronization.
+
+[NOTE]
+====
+These examples assume that all users are direct members of their respective
+groups. Specifically, no groups have other groups as members. See
+the Nested Membership Sync Example for information on
+how to sync nested groups.
+====
+
+In the RFC 2307 schema, both users (Jane and Jim) and groups exist on the LDAP
+server as first-class entries, and group membership is stored in attributes on
+the group. The following snippet of `ldif` defines the users and group for this
+schema:
+
+.LDAP entries that use RFC 2307 schema: *_rfc2307.ldif_*
+[source,ldif]
+----
+  dn: ou=users,dc=example,dc=com
+  objectClass: organizationalUnit
+  ou: users
+  dn: cn=Jane,ou=users,dc=example,dc=com
+  objectClass: person
+  objectClass: organizationalPerson
+  objectClass: inetOrgPerson
+  cn: Jane
+  sn: Smith
+  displayName: Jane Smith
+  mail: jane.smith@example.com
+  dn: cn=Jim,ou=users,dc=example,dc=com
+  objectClass: person
+  objectClass: organizationalPerson
+  objectClass: inetOrgPerson
+  cn: Jim
+  sn: Adams
+  displayName: Jim Adams
+  mail: jim.adams@example.com
+  dn: ou=groups,dc=example,dc=com
+  objectClass: organizationalUnit
+  ou: groups
+  dn: cn=admins,ou=groups,dc=example,dc=com <1>
+  objectClass: groupOfNames
+  cn: admins
+  owner: cn=admin,dc=example,dc=com
+  description: System Administrators
+  member: cn=Jane,ou=users,dc=example,dc=com <2>
+  member: cn=Jim,ou=users,dc=example,dc=com
+----
+<1> The group is a first-class entry in the LDAP server.
+<2> Members of a group are listed with an identifying reference as attributes on
+the group.
+
+To sync this group, you must first create the configuration file. 
+Then run the sync with the *_rfc2307_config.yaml_* file:
+
+----
+$ oc adm groups sync --sync-config=rfc2307_config.yaml --confirm
+----
+
+{product-title} creates the following group record as a result of the above sync
+operation:
+
+.{product-title} group created by using the *_rfc2307_config.yaml_* file
+[source,yaml]
+----
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+  annotations:
+    openshift.io/ldap.sync-time: 2015-10-13T10:08:38-0400 <1>
+    openshift.io/ldap.uid: cn=admins,ou=groups,dc=example,dc=com <2>
+    openshift.io/ldap.url: LDAP_SERVER_IP:389 <3>
+  creationTimestamp:
+  name: admins <4>
+users: <5>
+- jane.smith@example.com
+- jim.adams@example.com
+----
+<1> The last time this {product-title} group was synchronized with the LDAP server, in ISO 6801
+format.
+<2> The unique identifier for the group on the LDAP server.
+<3> The IP address and host of the LDAP server where this group's record is
+stored.
+<4> The name of the group as specified by the sync file.
+<5> The users that are members of the group, named as specified by the sync file.
+
+[[rfc2307-with-user-defined-name-mappings]]
+=== RFC2307 with user-defined name mappings
```

Comment:
I'm having a hard time following these headings and levels when I scroll down. It's hard to tell under which type (RFC, AD, augmented AD) the example I'm looking at is. 

So, is what's happening here is that you create the LDAP sync config file, and then show two different ways of syncing, "RFC2307 with user-defined name mappings" and "Syncing groups by using RFC 2307 with user-defined error tolerances". I think I would try to bump up the levels and separate these examples a bit better. Also the "Nested membership sync example" being under augmented AD and then its own one is confusing me.

Maybe instead of "LDAP group sync examples", you'd have one for each of the 3 at the top level, and then show the various ways of syncing under each?

Reply:
This was a stright copy-and-drop effort. I didn't want to mess with the material too much, as I do not know LDAP sync and didn't want to break anything. I will take another look.